### ISSUE TITLE
linter: Support negations in overrides

### ISSUE THREAD

**Issue by null77** [oxc-project/oxc#15932]
This syntax would be very useful in an override:

```
      "files": ["packages/services/**", "!packages/services/**/*.test.ts"],
      "rules": {
        "no-restricted-imports": [
          "error",
          {
            "patterns": [
              {
                "group": [
...snip...
```

Currently it doesn't seem to work.

**Comment by xuanduc987** [comment]
The override would be apply when a file match any pattern from `files` option, so to exclude some files we should add `ignores` option like eslint

**Comment by null77** [comment]
Yes, `ignores` would be a way to do this compatible with eslint and also without breaking existing behaviour. I would be happy with either, maybe the project owners can recommend which solution to use?

**Comment by rtarpine-vanta** [comment]
Without this, https://github.com/oxc-project/oxlint-migrate can't carry over the `overrides.ignores` fields in my ESLint config and I get lots of false positives when trying to migrate.

**Comment by AndreasArvidsson** [comment]
I would also like to upvote a ignores field like eslint has.

**Comment by EurFelux** [comment]
This issue prevents me from migrating from eslint to oxlint.

**Comment by camc314** [comment]
Will ship in the next release of oxlint (not the one releasing today) planned for 11th of May

**Comment by will-stone** [comment]
This has apparently been released, has anyone got it to work?

**Comment by camc314** [comment]
@will-stone this is undocumented for now as the behaviour may change. We wanted some further discussion within the team.

If you want to use it (at your own risk):
```
{
  "overrides: [
    {
      "files": ["packages/services/**"],
      "excludeFiles": ["!packages/services/**/*.test.ts"],
      "rules": {
        "no-restricted-imports": "error"
      }
    }
  ]
}
```
